### ISSUE TITLE
False positives in flow analysis found in Platform compatibility analyzer

### ISSUE THREAD

**Issue by buyaa-n** [dotnet/roslyn-analyzers#5938]
### False positives in flow analysis found in Platform compatibility analyzer

**Diagnostic ID**: [CA1416](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1416): Validate Platform compatibility`

### False positives in flow analysis found in Platform compatibility analyzer 

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100](https://dotnet.microsoft.com/download/dotnet/5.0) or later

#### Case 1: flow analysis false positives for nested if conditions
```cs
[SupportedOSPlatform("ios12.0")]
public class IosClass
{
    public void M1() { }
    [SupportedOSPlatform("ios13.0")]
    public void M2() { }
}

class Program
{
    void M3()
    {
        if (OperatingSystem.IsIOSVersionAtLeast(12, 0))
        {
            IosClass ios = new IosClass();
            if (OperatingSystem.IsIOSVersionAtLeast(13, 0))
            {
                ios.M2(); // Should not warn, but warns
            }
            ios.M1(); // Should not warn, but warns
        }
    }
}
```
#### Return statement with more than one condition combined with call site attribute produce false positivess
```cs
public class Test
{
    [SupportedOSPlatform("ios")]
    public void M1()
    {
        [|M2()|];

        if((OperatingSystem.IsIOS() && !OperatingSystem.IsIOSVersionAtLeast(13, 0)))
        {
            M2();
            return;
        }
        M3(); // should not warn as ios 13.0 or below case returns within above contions
    }

    [SupportedOSPlatform("ios")]
    [UnsupportedOSPlatform("ios13.0")]
    [SupportedOSPlatform("tvos")]
    [UnsupportedOSPlatform("tvos13.0")]
    public void M2() { }

    [SupportedOSPlatform("ios13.0")]
    [SupportedOSPlatform("tvos13.0")]
    public void M3() { }
}
```
Related to and found in https://github.com/dotnet/maui/issues/823

CC @mavasani @jeffhandley

**Comment by mattleibow** [comment]
I am worried about this issue as we had a crash recently because we missed a case where we should have had a check. However, since the analyzer has far too many false positives - most places in fact for iOS - we have it disabled.

**Comment by buyaa-n** [comment]
> However, since the analyzer has far too many false positives - most places in fact for iOS - we have it disabled.

There are 3 false positive cases mentioned here, which one is the main cause of the false positives in MAUI repo? Case 2 is already fixed, case 3 is not solvable as currently there is no way to get info about #if-defs in analysis context. Case 1 flow analysis result is very complex, but we can try to fix this case.
